chall11/cdorta-n.c

```diff
@@ -0,0 +1,197 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   cdorta-n.c                                         :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: cdorta-n <cdorta-n@student.42madrid.com    +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2020/05/21 13:42:55 by cdorta-n          #+#    #+#             */
+/*   Updated: 2020/05/21 21:44:23 by cdorta-n         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <stdio.h>
+#include <unistd.h>
+#include <stdlib.h>
+
+/*
+** 'Label' -> Positive Integer
+** 'goto' -> goto-statement
+** 'whitespaces' -> only 1 space
+** 'end-of-line' -> \n
+*/
+
+char	*ft_goto_error()
+{
+	char *error = "Infinite loop !";
+	return (error);
+}
+
+int		ft_goto_statement(const char *code, int num, int c)
+{
+		int i = 0;
+		int exp = 1;
+		int nums;
+		int aux;
+		int aux2;
+
+		nums = num;
+		aux = nums;
+		while (num > 9)
+		{
+			num /= 10;
+			exp *= 10;
+		}
+		aux2 = exp;
+		while (code[c] != '\0')
+		{
+			if (code[c] == (num + 48))
+			{
+				while (exp >= 10)
+				{
+					c++;
+					nums %= exp;
+					exp /= 10;
+					num = (nums / exp) + 48;
+					if (!(code[c] == num))
+					{
+						nums = aux;
+						exp = aux2;
+						num = nums / exp;
+						break ;
+					}
+				}
+				if (exp < 10)
+					return (c);
+			}
+			c++;
+		}
+		return (0);
+}
+
+int 	ft_goto_aux(const char *code)
+{
+	int c = 0;
+	int i = 0;
+	int num = 0;
+	
+	while (code[c] != '\0')
+	{
+		if (code[c] == 'g' && code[c + 1] == 'o' && code[c + 2] == 't' && code[c + 3] == 'o')
+		{
+			while (code[c] < '0' || code[c] > '9')
+				c++;
+			while (code[c] >= '0' && code[c] <= '9')
+			{
+				num = num * 10 + (code[c] - 48);
+				c++;
+			}
+			if ((ft_goto_statement(code, num, c) < c))
+				return (-1);
```

Comment:
This doesn't really work because the labels weren't always in incremental order. The order didn't really matter (there was an example about this).

A good way to check this was to record the previous labels that you have performed a jump to, and every time you are gonna perform a `goto`, you check if it's in the list.